aggregator/src/aggregator/aggregation_job_creator.rs

```diff
@@ -361,77 +353,58 @@ impl<C: Clock + 'static> AggregationJobCreator<C> {
             .run_tx_with_name("aggregation_job_creator_time_no_param", |tx| {
                 let (this, task) = (Arc::clone(&self), Arc::clone(&task));
                 Box::pin(async move {
-                    let current_batch_start = this
-                        .clock
-                        .now()
-                        .to_batch_interval_start(task.time_precision())?;
-
-                    // Find some unaggregated client reports, and group them by their batch.
-                    let report_ids_by_batch = tx
+                    // Find some unaggregated client reports.
+                    let report_ids_and_times = tx
                         .get_unaggregated_client_report_ids_for_task(task.id())
-                        .await?
-                        .into_iter()
-                        .map(|(report_id, time)| {
-                            time.to_batch_interval_start(task.time_precision())
-                                .map(|rounded_time| (rounded_time, (report_id, time)))
-                                .map_err(datastore::Error::from)
-                        })
-                        .collect::<Result<Vec<_>, _>>()?
-                        .into_iter()
-                        .into_group_map();
+                        .await?;
 
                     // Generate aggregation jobs & report aggregations based on the reports we read.
                     let mut agg_jobs = Vec::new();
                     let mut report_aggs = Vec::new();
-                    for (batch_start, report_times_and_ids) in report_ids_by_batch {
-                        let batch_interval = Interval::new(batch_start, *task.time_precision())?;
-                        for agg_job_reports in
-                            report_times_and_ids.chunks(this.max_aggregation_job_size)
-                        {
-                            if batch_start >= current_batch_start
-                                && agg_job_reports.len() < this.min_aggregation_job_size
-                            {
-                                continue;
-                            }
+                    for agg_job_reports in
+                        report_ids_and_times.chunks(this.max_aggregation_job_size)
+                    {
+                        if agg_job_reports.len() < this.min_aggregation_job_size {
```

Comment:
This is a bit stricter than the previous behavior about minimum aggregation job sizes, since there's no longer an exception for the current batch. It seems our integration tests are okay since they run this with `min_aggregation_job_size: 1`. I think the overhead benefits of this are good enough that we should probably keep it, and accept more tail latency on aggregation of low-volume tasks.

Reply:
I agree; the new behavior is that we'll wait until we have at least `min_aggregation_job_size` reports available before we generate an aggregation job. This means that low-volume tasks will see increased latency. I agree that we should keep it for now -- if this behavior ends up being problematic, we could eventually add logic to notice if reports are "too old" & generate a smaller-than-min-size aggregation job in this case.